src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/Nuget/ProjectRestoreInfoBuilder.cs

```diff
@@ -0,0 +1,58 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Immutable;
+using System.Linq;
+using NuGet.SolutionRestoreManager;
+
+namespace Microsoft.VisualStudio.ProjectSystem.VS.Nuget
+{
+    internal static class ProjectRestoreInfoBuilder
+    {
+        internal static IVsProjectRestoreInfo Build(ImmutableList<IProjectValueVersions> updates)
+        {
+            string baseIntermediatePath = null;
+            var targetFrameworks = new TargetFrameworks();
+            
+            foreach (IProjectVersionedValue<IProjectSubscriptionUpdate> update in updates)
+            {
+                var configurationChanges = update.Value.ProjectChanges[ConfigurationGeneral.SchemaName];
+                baseIntermediatePath = baseIntermediatePath ?? configurationChanges.After.Properties["BaseIntermediateOutputPath"];
+                string targetFrameworkMoniker = configurationChanges.After.Properties["TargetFrameworkMoniker"];
+
+                if (targetFrameworks.Item(targetFrameworkMoniker) == null)
+                {
+                    var projectReferencesChanges = update.Value.ProjectChanges[ProjectReference.SchemaName];
+                    var packageReferencesChanges = update.Value.ProjectChanges[PackageReference.SchemaName];
+
+                    targetFrameworks.Add(new TargetFrameworkInfo
+                    {
+                        TargetFrameworkMoniker = targetFrameworkMoniker,
+                        ProjectReferences = GetReferences(projectReferencesChanges.After.Items),
```

Comment:
This logic looks wrong - it looks like all "TFMs" get exactly the same set of project references and package references, so if I have a condition that excludes a package or a project from a TFM, it will still be included. Given the data is combined - I'm not sure of a good way to figure that out.